cmake/spheral/InstantiateCXX.cmake

```diff
@@ -15,48 +15,65 @@ function(instantiate _inst_var _source_var)
 
   # Create our list of dimension to instantiate
   set(_dims )
+
   if(ENABLE_1D)
      list(APPEND _dims 1)
   endif()
+
   if(ENABLE_2D)
      list(APPEND _dims 2)
   endif()
+
   if(ENABLE_3D)
      list(APPEND _dims 3)
   endif()
 
   # Iterate over each Instantation file
   foreach(_inst ${${_inst_var}})
-
     if(ENABLE_INSTANTIATIONS)
-
-      # Generate a C++ file for each dimension with the format: <Name>Inst<N>d.cc
-      foreach(_dim ${_dims})
-
+      if(SPHERAL_COMBINE_INSTANTIATIONS)
+        # Generate a C++ file with the format: <Name>Inst.cc
         set(_inst_py ${CMAKE_CURRENT_SOURCE_DIR}/${_inst}Inst.cc.py)
-        set(_inst_file ${_inst}Inst${_dim}d.cc)
+        set(_inst_file ${_inst}Inst.cc)
 
         # Generate the C++ file
         # Uses BLT's python for instantiations to work when building CXX_ONLY as well as with python
         add_custom_command(OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/${_inst_file}
                            DEPENDS ${_inst_py}
-                           COMMAND ${Python3_EXECUTABLE} ${SPHERAL_ROOT_DIR}/src/helpers/InstantiationGenerator.py ${_inst_py} ${_inst_file} ${_dim}
+                           COMMAND ${Python3_EXECUTABLE} ${SPHERAL_ROOT_DIR}/src/helpers/InstantiationGenerator.py ${_inst_py} ${_inst_file} ${_dims}
                            BYPRODUCTS ${_inst_file}
                            COMMENT "Generating instantiation ${_inst_file}..."
                            )
 
         # Add the instantiation files to the sources
         list(APPEND _tmp_source ${_inst_file})
-      endforeach()
+      else()
```

Comment:
Is there any reason to even make this optional? @jmikeowen @adayton1

Reply:
I was thinking of having it be an option for period of time so that if any unforeseen issues arose you would have an easy switch. After it has been exercised in anger for a bit, we could remove the option.